pkg/cloud/cloud.go

```diff
@@ -235,6 +235,8 @@ func newEC2Cloud(region string) (Cloud, error) {
 	awsConfig := &aws.Config{
 		Region:                        aws.String(region),
 		CredentialsChainVerboseErrors: aws.Bool(true),
+		// Set MaxRetries to a high value. It will be "ovewritten" if context deadline comes sooner.
+		MaxRetries: aws.Int(8),
```

Comment:
for my reference: https://github.com/aws/aws-sdk-go/blob/d87d92b4412739c6ef8917979e922fae98b7654c/aws/client/default_retryer.go

8 make sense to me. I played with value here:  https://play.golang.org/p/sZr9Tgxgt79   SDK jitters the delay every retry but on average 8 retries is the right amount to assuming  a 1 min polling period

Reply:
Since this affects all api calls, i also checked attach detach.

By default, the attacher passes in a context of 15 seconds: https://github.com/kubernetes-csi/external-attacher/blob/3d49e838d8ecc491b0a7ce423f32b7e43aaa51af/cmd/csi-attacher/main.go#L59

So 3 retries was too low even for that.

Reply:
Sorry, I'm mixing up the sdk-level retries and the driver-level retries.

So it's really sdk-level retry * driver-level retry.

Either way, the point is that we need to increase sdk-level retry so that the driver-level retry doesn't prematurely exit due to a throttling error